ceremony/2022-09-27/staged/root.json

```diff
@@ -0,0 +1,164 @@
+{
+	"signed": {
+		"_type": "root",
+		"spec_version": "1.0",
+		"version": 5,
+		"expires": "2023-03-27T22:43:28Z",
+		"keys": {
+			"25a0eb450fd3ee2bd79218c963dce3f1cc6118badf251bf149f0bd07d5cabe99": {
+				"keytype": "ecdsa-sha2-nistp256",
+				"scheme": "ecdsa-sha2-nistp256",
+				"keyid_hash_algorithms": [
+					"sha256",
+					"sha512"
+				],
+				"keyval": {
+					"public": "-----BEGIN PUBLIC KEY-----\nMFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAEEXsz3SZXFb8jMV42j6pJlyjbjR8K\nN3Bwocexq6LMIb5qsWKOQvLN16NUefLc4HswOoumRsVVaajSpQS6fobkRw==\n-----END PUBLIC KEY-----\n"
```

Comment:
My understanding was that we wouldn't produce PEM-encoded keys for a little while longer.

Reply:
> I'd rather not push it out after GA: without it, we still aren't TUF spec compatible.

I totally agree.

> I could backport sigstore versions, but we don't have release version branches in sigstore right now

I don't think that fixes it. I'm less concerned about people who are unwilling to update to a new major version (though [that sometimes happens for good reasons](https://github.com/fluxcd/source-controller/issues/899)) and more concerned about people who just...haven't updated it.

> I'm not sure why people wouldn't update

2 weeks is a pretty aggressive timeline IMO. I wonder whether we have any user agent data on this.

For instance, looks like the [Arch package is still at `v1.11.1`](https://archlinux.org/packages/community/x86_64/cosign/). Also, people pin their dependencies and don't update immediately: for instance, [the v1.11.1 GH action](https://github.com/search?q=14d43345ff50608baaa37893f4822c406ed470a9&type=code), [v1.11.0](https://github.com/search?q=8d0fee40fd011ad052c0fed30c58d35203e9d130&type=code), etc. I personally have gone >2 weeks between `apt-get updates` 🤷 

----------------------------------------------

I'm not trying to put my foot down and stop this change, I just want to make sure it's explicitly called out and acknowledged by the keyholders and maybe the TSC before merge; right now, the PR description doesn't acknowledge that it's a breaking change.

I'd also like to point out a few alternatives. For instance, we could maintain two parallel TUF repos and keep producing old-format keys for a little while longer while pointing new Cosign builds at the new repo by default. That'd buy us some time to switch.

Reply:
> . I wonder whether we have any user agent data on this.

Yeah! I have thought about this: I thought about creating mirrors for each major version so we can detect which versions of sigstore people are using to access TUF. From the TUF root perspective, since it's a public GCS bucket we can't get any monitoring directly on that though.

> I'd also like to point out a few alternatives. For instance, we could maintain two parallel TUF repos and keep producing old-format keys for a little while longer while pointing new Cosign builds at the new repo by default. That'd buy us some time to switch.

Wouldn't this introduce issues with rollback attacks? I could try "splitting" repositories, but then merging them again would be an issue: clients on the old repository who want to migrate to the new one are going to have issues when they realize the versions of the snapshot did not increment necessarily.

Reply:
I thought fwiw I gave keyholders and TSC notice of this :/

Regarding updates: short of manually scraping and updating, I can post an advisory notice in sigstore as a reference issue anticipating the problem for people who have not bumped or have dependabot?

Reply:
@shibumi: Wasn't sure of a better channel to reach you on, since I can't post issues here. Any reason why arch is still distributing 1.11? I see https://github.com/archlinux/svntogit-community/commit/b50eee28a63e5f11b61b5cfed9f22dbc42e9f0f2 bumped it

Reply:
> From the TUF root perspective, since it's a public GCS bucket we can't get any monitoring directly on that though.

Oh, that's annoying. I wonder if we moved it over to a proper CDN....but that can be a "for later" thing.

> Wouldn't this introduce issues with rollback attacks? I could try "splitting" repositories, but then merging them again would be an issue: clients on the old repository who want to migrate to the new one are going to have issues when they realize the versions of the snapshot did not increment necessarily.

It kinda sounds like TAP-14 to be honest—I don't think there are rollback issues here, you should be able to cut over whenever. I think it just postpones the problem, to be clear. But we can keep doing it indefinitely.

Sounds like a big pain though.

> I thought fwiw I gave keyholders and TSC notice of this :/

You totally might have! I'm just in neither category so I didn't see it 🙂 And I didn't see it on the PR either, just trying to err on the side of overcommunication.

Reply:
@asraa sorry, totally missed your notification, because I just started a new job at Google :)

I updated cosign over 10 days ago: 2022-09-29 15:08 UTC. So this is fixed, I would assume :)